Core/BaseObject/ObjectRepository.php

```diff
@@ -42,25 +42,86 @@ public function __construct()
      * 
      * @param $user Core\BaseObject\BaseObject
      */
-    abstract public function create($user);
+    public function create($entity)
+    {
+        // $entityKeyNames holds entity attrbute names  
+        $entityKeyNames = array_keys($entity->getAttributes());
+
+        // $entityKeyValues holds entity attrbute values 
+        $entityKeyValues = array_values($entity->getAttributes());
+
+        // Database inser prepare 
+        $query = $this->db->DatabaseConnection->exec('INSERT INTO ' . $entity->tableName . " VALUES ('id','" . implode("','" , $entityKeyValues) . "')");
+        if(!$query)
+        {
+            var_dump($this->db->DatabaseConnection->errorInfo());
+        }
+        // Clears query
+        $query = "";
+    }
 
     /**
-     * function read()
+     * function findById()
      * 
      * holds logic for reading new instance of Core\BaseObject\BaseObject
      * 
-     * @param $id ID of Core\BaseObject\BaseObject
+     * @param $attributes assosiative array [ attributes[{Child of BaseObject} tableName] , attributes[{Child of BaseObject} primaryKeyName] ] 
      */
-    abstract public function read($id);
+    public function findById($attributes)

```

Comment:
this must be only an id

```suggestion
    public function findById($id)
 // just past down the id
```